### ISSUE TITLE
Function `inLogs` doesn't need to be declared `async`

### ISSUE THREAD

**Issue by barakman** [OpenZeppelin/openzeppelin-solidity#1167]
## 🎉 Description

File [expectEvent.js](https://github.com/OpenZeppelin/openzeppelin-solidity/edit/master/test/helpers/expectEvent.js):

Function `inLogs` doesn't need to be declared `async`.

Then, anyone calling it doesn't need to `await` for completion.

**Comment by nventuro** [comment]
I wonder why it was made to return a promise in the first place, maybe something changed and we forgot to remove that `async`? Anyway, thanks a lot @barakman!

**Comment by barakman** [comment]
@nventuro:

Thank you.
Not critical, but is this going to be fixed soon?

Also, not directly related, but I'm not sure where else to ask this:
When is the next formal release expected, and where can I generally find future release dates and versions?

We need to do our own release soon, and since OpenZeppelin code is embedded mostly into our two Token contracts, it will not be upgradable on our system (can replace "inner" contracts, but not the ones which hold the actual balance).

So we would ideally like to have a verified version of OpenZeppelin (and even more ideally, a x.0.0 kind of version).

Thank you, and I apologize for posting this possibly in the wrong place.

**Comment by nventuro** [comment]
I've opened a PR fixing this (#1169), you can see it on this very same issue.

The next release (1.12) is actually coming out today, and the next version after that is 2.0, coming out in about a month. You can look at the 2.0 milestone and related issues/PRs [here](https://github.com/OpenZeppelin/openzeppelin-solidity/milestone/3). There will probably be a blogpost soon describing what 2.0 entails.

That said, OpenZeppelin code has been around for quite a bit and is used by multiple projects, so I'm not sure what sort of extra 'verification' you're expecting. Small issues like this don't really affect the smart contract code, but rather the maintainability of the project.

By the way, if you're interested in upgradeability, take a look at [ZeppelinOS](https://zeppelinos.org/), which tackles that issue head-on.

**Comment by nventuro** [comment]
There will be some API-breakage in 2.0, but no current functionality will be lost. Regarding official releases, you should always use the releases bundled up in npm (i.e. `npm install openzeppelin-solidity`), as opposed to downloading the code straight from the development branch. 1.11 and 1.12 are both official releases, and should serve your purposes just fine.

**Comment by barakman** [comment]
Ahh, I see that those recent changes that I just mentioned are not there yet.
This means that I am going to have to rewind all my recent updates.
Any chance they are included in the upcoming release today?

Also, in general, you suggestion for `npm install openzeppelin-solidity` is somewhat problematic for me.
My ecosystem is based on Truffle, which expects to find all contracts under the same (`contracts`) folder.
This means that to begin with, I will need to add a `postinstall` script to my `package.json` file, which will copy your contracts from folder `/node_modules/openzeppelin-solidity/contracts` into folder `/contracts`.
Second, we use only a subset of your contracts, but this is minor.
Third, we use a modified version of your `MintableToken` contract (so again, a `postinstall` script is required).

By the way, `npm install` does not download your [mocks](https://github.com/OpenZeppelin/openzeppelin-solidity/tree/master/contracts/mocks) folder, so I will not even be able to run your tests.

**Comment by nventuro** [comment]
There is no need for you to run our tests locally, since they've been run for each single build. 

Truffle also **will** recognize modules installed with npm (it is our main usage scenario), you just need to do e.g. `import "openzeppelin-zos/contracts/token/ERC20/ERC20.sol";`.

Regarding the `MintableToken` change, I would recommend against modifying OpenZeppelin code, and doing the changes you require via inheritance, if at all possible. What is it about the contract that doesn't suit your project's needs?

**Comment by barakman** [comment]
@nventuro:

I use `truffle` for testing, but `solc` for compiling (i.e., for deploying eventually).
So if I understand your suggestion correctly, then I would be forced to use different import paths inside my contracts (one for `truffle` and one for `solc`).
And that would require some sort of project configuration within an IDE (which I am currently not using, and I'm not even aware of any such tool for Solidity, to be honest).
Am I wrong?

With regards to the `Mintable` contract, we changed the `mint` function from:
    hasMintPermission canMint public
To:
    internal
This is because we have our own internal mechanism for minting at certain points.

By the way, `npm install` does not download your [mocks](https://github.com/OpenZeppelin/openzeppelin-solidity/tree/master/contracts/mocks) folder.

**Comment by barakman** [comment]
@nventuro: Hi

I have recently come across the requirement to install OpenZeppelin code via `npm` (instead of manually copying all required source files, as I previously described in this thread).

This goes back to your suggestion.
At the time, I noted to you that Truffle had a "hard time" dealing with files outside of the `contracts` folder.

However, I would like to correct that description of mine, as `truffle compile` works just fine.
All I need to change is the relative path that I use for importing OpenZeppelin files in my contracts (prepending it with `../node_modules/openzeppelin-solidity/contracts`).

It is in fact that Solidity compiler (`solc`) which cannot deal with "external" files.

Now, `solc` has a compilation switch called `--allow-paths`, which is supposed to allow one to import files from outside the working folder.
Unfortunately, this switch is currently not functioning correctly, and `solc` keeps throwing multiple "File outside of allowed directories" errors, regardless of the value that you set it to.

I am working in collaboration with a contributor in order to resolve this (see [here](https://github.com/ethereum/solidity/issues/2742#issuecomment-426240873)), but until it is fixed, I have been wondering:
How do other people compile and deploy their contracts? Does everybody do it with Truffle?
Because clearly, if they use the Solidity compiler, then they must copy OpenZeppelin files into their project, in the same way that I do it.

Would you know of any way to combine the installation of OpenZeppelin via `npm`, and the compilation of the contracts via `solc`?

Thanks

**Comment by nventuro** [comment]
I don't have any hard data on this, but would expect most people to use `truffle` (or a very similar alternative). The main reason behind this is not so much the compilation itself, but the need to develop tests for your contracts: `truffle` comes integrated with `ganache-cli`, `mocha`, web3, and provides the `truffle-contracts` wrapper, so doing this is remarkably simple. 

Is there a specific reason why you want to call `solc` directly from the command line? From what I understand, you used to simply copy the code over, and that worked. What about testing?

**Comment by barakman** [comment]
@nventuro:

Thank you for your quick response.

I use:
1. `truffle test` (preceded by `truffle compile`) along with `ganache-cli` for testing.
2. `solidity-coverage` along with `testrpc-sc` for a full coverage report.
3. The output of `solc` (abi and bin files) for contract-deployment and contract-interaction via web3.js v1.x directly (in opposed to the previous two, where web3.js v0.x is used indirectly).

For #1 and #2, I even implemented a NodeJS script which starts the Ethereum node (either `ganache-cli` or `testrpc-sc`) and then runs the test (either  `truffle test` or `solidity-coverage`). The script is connected in the `package.json` file, so all I need to do after installation is `npm test 1` or `npm test 2`.
This script also has the advantage of being OS-agnostic (platform-independent).

Why did I choose to use `solc` for the actual deployment and interaction?

Well, first and foremost because it generates bin and abi files, while Truffle generates json files.
And since I would ultimately need to publish the abi files, I've figured that I may as well go along with `solc` for this purpose.
Someone insinuated to me earlier today that I could extract bin and abi from the json, but why should I bother with having to "auto-script" that?

Second, as you mentioned, Truffle testing relies on the Mocha framework. I could use it also for contract interaction, but that's not the purpose of Mocha, is it?
Truffle deployment (via `truffle deploy` or `truffle migrate`) does allow me to deploy the contracts without going via the Mocha framework. However, I have found their deployment syntax to be very cumbersome, and not very convenient for use.

Third, Truffle is still stuck on web3.js v0.18.4 for some reason.
